discord/utils.py

```diff
@@ -1111,7 +1111,12 @@ async def autocomplete_callback(ctx: AutocompleteContext) -> V:
         if asyncio.iscoroutine(_values):
             _values = await _values
 
-        gen = (val for val in _values if str(val).lower().startswith(str(ctx.value or "").lower()))
+        def check(item: Any) -> bool:
+            if hasattr(item, "to_dict"):
+                item = item.to_dict()["name"]
```

Comment:
```suggestion
            if hasattr(item, "to_dict"):
                item = item.to_dict()["name"]
```
Why not check for `.name` here?